internal/controlplane/handlers_repositories.go

```diff
@@ -312,84 +313,47 @@ func (s *Server) GetRepositoryByName(ctx context.Context,
 	return &pb.GetRepositoryByNameResponse{Repository: r}, nil
 }
 
-// DeleteRepositoryById deletes a repository by name
-func (s *Server) DeleteRepositoryById(ctx context.Context,
-	in *pb.DeleteRepositoryByIdRequest) (*pb.DeleteRepositoryByIdResponse, error) {
+// DeleteRepositoryById deletes a repository by its UUID
+func (s *Server) DeleteRepositoryById(
+	ctx context.Context,
+	in *pb.DeleteRepositoryByIdRequest,
+) (*pb.DeleteRepositoryByIdResponse, error) {
 	parsedRepositoryID, err := uuid.Parse(in.RepositoryId)
 	if err != nil {
 		return nil, util.UserVisibleError(codes.InvalidArgument, "invalid repository ID")
 	}
 
-	projectID := getProjectID(ctx)
-	provider, err := getProviderFromRequestOrDefault(ctx, s.store, in, projectID)
-	if err != nil {
-		return nil, providerError(err)
-	}
-
-	// read the repository
-	repo, err := s.store.GetRepositoryByIDAndProject(ctx, db.GetRepositoryByIDAndProjectParams{
-		ID:        parsedRepositoryID,
-		ProjectID: projectID,
+	err = s.deleteRepository(ctx, in, func(client v1.GitHub, projectID uuid.UUID) error {
+		return s.repos.DeleteRepositoryByID(ctx, client, projectID, parsedRepositoryID)
 	})
-	if errors.Is(err, sql.ErrNoRows) {
-		return nil, status.Errorf(codes.NotFound, "repository not found")
-	} else if err != nil {
-		return nil, status.Errorf(codes.Internal, "cannot read repository: %v", err)
-	}
-
-	err = s.deleteRepositoryAndWebhook(ctx, repo, provider)
 	if err != nil {
 		return nil, err
 	}
 
-	// Telemetry logging
-	logger.BusinessRecord(ctx).Provider = repo.Provider
```

Comment:
Telemetry now lives in the service layer. This is because some of the attributes being tracked are only available inside the service (e.g. the ID of the repo when deleting by name). I considered returning the data needed for telemetry, but this made the abstraction very leaky.

Reply:
We should be extracting the telemetry from the request earlier than we were anyway -- right now most of our errors don't include much telemetry about the resources involved.